pos_lot_selection/static/src/js/popup/OrderWidget.js

```diff
@@ -0,0 +1,47 @@
+odoo.define('pos_lot_selection.CustomOrderWidget', function(require) {
+    'use strict';
+
+    const Registries = require('point_of_sale.Registries');
+    const OrderWidget = require('point_of_sale.OrderWidget');
+
+    
+    var CustomOrderWidget = (OrderWidget) => class extends OrderWidget{
+        async _editPackLotLines(event) {
+            const orderline = event.detail.orderline;
+            
+            this.env.session.lots = await this.get_lots(orderline.product, this.env.session);
```

Comment:
I'm not sure if the session object is the right place where to store the possible values. Maybe the POS DB? @ivantodorovich any better idea?

Reply:
EditListPopup is synchronous so to be able to fetch products there we need to load lots on session start, that implementation was declined 
second, is that there is no product in EditListPopup, so I need somehow to send the product there, or override part of the method, but that brings another issue as default implementation can call a chain of popups to configure the product and get a weight and we have an issue because lot popup is in the middle of that chain